src/libcore/mem.rs

```diff
@@ -133,26 +188,41 @@ pub fn size_of<T>() -> usize {
     unsafe { intrinsics::size_of::<T>() }
 }
 
-/// Returns the size of the given value in bytes.
+/// Returns the size of the pointed-to value in bytes.
+///
+/// This is usually the same as `size_of::<T>()`. However, when `T` *has* no
+/// statically known size, e.g. a slice [`[T]`][slice] or a [trait object][],
+/// then `size_of_val` can be used to get the dynamically-known size.
+///
+/// [slice]: ../primitive.slice.html
+/// [trait object]: ../../book/trait-objects.html
 ///
 /// # Examples
 ///
 /// ```
 /// use std::mem;
 ///
 /// assert_eq!(4, mem::size_of_val(&5i32));
+///
+/// let x: [u8; 13] = [0; 13];
+/// let y: &[u8] = &x;
+/// assert_eq!(13, mem::size_of_val(y));
 /// ```
 #[inline]
 #[stable(feature = "rust1", since = "1.0.0")]
 pub fn size_of_val<T: ?Sized>(val: &T) -> usize {
     unsafe { intrinsics::size_of_val(val) }
 }
 
-/// Returns the ABI-required minimum alignment of a type
+/// Returns the [ABI][]-required minimum alignment of a type.
+///
+/// Every valid address of a value of the type `T` must be a multiple of this number.
```

Comment:
Really? I think "pointer to" is fine but "address to" sounds weird to me. I chose "address" because it focuses on the numerical value of the pointer, rather than the pointer itself, which is a more abstract type.